dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -232,61 +249,54 @@ void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring
         &hKey,
         nullptr /* lpdwDisposition */));
 
-    // Retrieve DisplayName and IconUri
-    // - DisplayName: Retrieve from Shell. If not specified, fall back to filename.
-    // - Icon: Retrieve from Shell. If it's not the case or the file extension is unsupported, then throw.
-    winrt::com_ptr<IPropertyStore> propertyStore;
-    wil::unique_hwnd hWindow{ GetConsoleWindow() };
-    THROW_IF_FAILED(SHGetPropertyStoreForWindow(hWindow.get(), IID_PPV_ARGS(propertyStore.put())));
-
     // Retrieve the display name
     std::wstring displayName{};
+    displayName = SetDisplayNameBasedOnProcessName();
 
-    wil::unique_prop_variant propVariantDisplayName;
-    // Do not throw in case of failure. We can use the filepath approach below as fallback to set a DisplayName.
-    LOG_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchDisplayNameResource, &propVariantDisplayName));
-    if (propVariantDisplayName.vt == VT_LPWSTR && propVariantDisplayName.pwszVal != nullptr)
-    {
-        displayName = propVariantDisplayName.pwszVal;
-    }
+    // Retrieve the icon
+    std::wstring iconFilePath{};
 
-    if (displayName.length() == 0)
+    wil::unique_hwnd hWindow{ GetConsoleWindow() };
+
+    if (hWindow)
     {
-        THROW_IF_FAILED(wil::GetModuleFileNameExW(GetCurrentProcess(), nullptr, displayName));
+        // Retrieve DisplayName and IconUri
+        // - DisplayName: Retrieve from Shell. If not specified, fall back to filename.
+        // - Icon: Retrieve from Shell. If it's not the case or the file extension is unsupported, then throw.
+        winrt::com_ptr<IPropertyStore> propertyStore;
+        THROW_IF_FAILED(SHGetPropertyStoreForWindow(hWindow.get(), IID_PPV_ARGS(propertyStore.put())));
+
+        wil::unique_prop_variant propVariantDisplayName;
+        // Do not throw in case of failure. We can use the filepath approach below as fallback to set a DisplayName.
+        THROW_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchDisplayNameResource, &propVariantDisplayName));
+        if (propVariantDisplayName.vt == VT_LPWSTR && propVariantDisplayName.pwszVal != nullptr)
+        {
+            displayName = propVariantDisplayName.pwszVal;
```

Comment:
check the length of pwszval to make sure it isnt zero else throw e_unexpected

Reply:
pavan why check for VT_EMPTY? We know in our success case it has to be VT_LPWSTR if vt isn't set to VT_LPWSTR we just default to filepath name. So I think we don't need any checks here I guess. Let me know.

Reply:
Setting an empty VT_LPWSTR results in getting a VT_EMPTY. I think such check will be redundant.

Reply:
Yeah the only thing here is, that the document isn't clear that if user set an empty string does it return a VT_EMPTY. It only says that:

If the [PROPERTYKEY](https://msdn.microsoft.com/en-us/library/bb773381(v=vs.85)) referenced in key is not present in the property store, this method returns S_OK and the vt member of the structure pointed to by pv is set to VT_EMPTY."

I think pavan might be right here that: propVariantDisplayName.vt == VT_LPWSTR && propVariantDisplayName.pwszVal != nullptr && propVariantDisplayName.pwszVal != L""

@danielayala94 any thoughts????

Reply:
The only way the app could reach this is if they explicitly set the propVariantDisplayName.vt to VT_LPWSTR. Unless I am missing something like default values for hWindow property store, the user is trying to set the property store. 

If in this flow they decide to use L"" for their displayName, I think it is a valid case. It seems intentional since pwszVal is nullptr by default and if they set it to empty string, well, that's what they want.

But is it really what they want? I don't know. :) but my thoughts are allowing the developer to do this since it looks like a conscious decision.

Reply:
I tried setting an empty VT_LPWSTR in a dummy Win32 app. That's how I came up with my statement above.